connector/src/main/java/com/google/cloud/hive/bigquery/connector/output/indirect/IndirectUtils.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hive.bigquery.connector.output.indirect;
+
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.hive.bigquery.connector.Constants;
+import com.google.cloud.hive.bigquery.connector.config.HiveBigQueryConfig;
+import com.google.cloud.hive.bigquery.connector.utils.HiveUtils;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.TaskAttemptID;
+import repackaged.by.hivebqconnector.com.google.common.collect.ImmutableList;
+
+public class IndirectUtils {
+
+  static Pattern gcsUriPattern = Pattern.compile("gs://([^/]*)(.*)?");
+
+  /** Retrieves the bucket name from a fully-qualified GCS URI. */
+  public static String extractBucketNameFromGcsUri(String gcsURI) {
+    Matcher m = gcsUriPattern.matcher(gcsURI);
+    if (m.find()) {
+      return m.group(1);
+    } else {
+      throw new RuntimeException("Incorrect GCS URI: " + gcsURI);
+    }
+  }
+
+  /** Returns true if the logged-in user has access to the given GCS path. */
+  public static boolean hasGcsWriteAccess(String gcsURI) {
+    String bucket = extractBucketNameFromGcsUri(gcsURI);
+    Storage storage = StorageOptions.newBuilder().build().getService();
```

Comment:
This takes the default account. What happens if the customer has set their own credentials?

Reply:
Good question. I suppose I should probably use `BigQueryCredentialsSupplier`, right?

Reply:
Ok, I've updated various parts of the code to use `BigQueryCredentialsSupplier` when interacting with GCS as well. That way there's a single source of credentials for everything.